databricks/sdk/dbutils.py

```diff
@@ -189,128 +199,34 @@ def widgets(self):
-                          context_factory=self._context_factory,
-                          util=self._name,
-                          method=method)
-
-
-import html
-import re
+        return _ProxyCall(self._executor, self._name, method)
 
 
 class _ProxyCall:
 
-    def __init__(self, *, command_execution: compute.CommandExecutionAPI,
-                 context_factory: typing.Callable[[], compute.ContextStatusResponse], cluster_id: str,
-                 util: str, method: str):
-        self._commands = command_execution
-        self._cluster_id = cluster_id
-        self._context_factory = context_factory
+    def __init__(self, executor: compute_ext.CommandExecutor, util: str, method: str):
+        self._executor = executor
         self._util = util
         self._method = method
 
-    _out_re = re.compile(r'Out\[[\d\s]+]:\s')
-    _tag_re = re.compile(r'<[^>]*>')
-    _exception_re = re.compile(r'.*Exception:\s+(.*)')
-    _execution_error_re = re.compile(
-        r'ExecutionError: ([\s\S]*)\n(StatusCode=[0-9]*)\n(StatusDescription=.*)\n')
-    _error_message_re = re.compile(r'ErrorMessage=(.+)\n')
-    _ascii_escape_re = re.compile(r'(\x9B|\x1B\[)[0-?]*[ -/]*[@-~]')
-
-    def _is_failed(self, results: compute.Results) -> bool:
-        return results.result_type == compute.ResultType.ERROR
-
-    def _text(self, results: compute.Results) -> str:
-        if results.result_type != compute.ResultType.TEXT:
-            return ''
-        return self._out_re.sub("", str(results.data))
-
-    def _raise_if_failed(self, results: compute.Results):
-        if not self._is_failed(results):
-            return
-        raise DatabricksError(self._error_from_results(results))
-
-    def _error_from_results(self, results: compute.Results):
-        if not self._is_failed(results):
-            return
-        if results.cause:
-            _LOG.debug(f'{self._ascii_escape_re.sub("", results.cause)}')
-
-        summary = self._tag_re.sub("", results.summary)
-        summary = html.unescape(summary)
-
-        exception_matches = self._exception_re.findall(summary)
-        if len(exception_matches) == 1:
-            summary = exception_matches[0].replace("; nested exception is:", "")
-            summary = summary.rstrip(" ")
-            return summary
-
-        execution_error_matches = self._execution_error_re.findall(results.cause)
-        if len(execution_error_matches) == 1:
-            return "\n".join(execution_error_matches[0])
-
-        error_message_matches = self._error_message_re.findall(results.cause)
-        if len(error_message_matches) == 1:
-            return error_message_matches[0]
-
-        return summary
-
     def __call__(self, *args, **kwargs):
         raw = json.dumps((args, kwargs))
         code = f'''
         import json
         (args, kwargs) = json.loads('{raw}')
         result = dbutils.{self._util}.{self._method}(*args, **kwargs)
-        dbutils.notebook.exit(json.dumps(result))
```

Comment:
Does `dbutils.notebook.exit` work with `print`? If so, maybe we can convert `return` to `dbutils.notebook.exit(json.dumps(...))` so that you don't have to fail if other print statements are included?

Reply:
Alternatively, we can take the last line of output, since print commands before will always add newlines?